### PR TITLE
Make NGV rule over existing mods, even from specific models

### PR DESCRIPTION
**Pull request by ferdonline**

### CODE REVIEW

File: var/spack/repos/builtin/packages/neurodamus-model/package.py [hunk 1]
```diff
@@ -83,12 +83,13 @@ def setup_common_mods(self, spec, prefix):
             shutil.move('common', '_common_orig')
             force_symlink(spec.variants['common_mods'].value, 'common')
 
-        if spec.satisfies("+ngv"):
-            copy_all("ngv/common/mod/ngv", "mod")
-
     def build_model(self, spec, prefix):
         """Build and install the bare model.
         """
+        # NGV must overwrite other mods, even from the specific models, e.g. ProbAMPANMDA
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/neurodamus-model/package.py]: I assume this is >= 80 column.

**Review by pramodk — DISMISSED** [review]
Apart from style fix, LGTM

**Comment by pramodk** [comment]
For this PR, we are seeing new error from pip check introduced in #1189:

```console
[2021-06-04T13:02:42.821Z] Successfully installed pip-21.1.2 setuptools-57.0.0 wheel-0.36.2
[2021-06-04T13:02:47.252Z] + make -k -j 72 -f /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/1187/tmp/tmp.YFvZB3EhyM boost/1.70.0 boost/1.73.0 neuron/8.0.0 py-cgal-pybind/0.1.0 py-dask-mpi/2.21.0 py-flake8/3.8.2 py-h5py/3.1.0 py-mpi4py/3.0.3 py-numpy/1.19.4 py-poisson-recon-pybind/0.1.0 py-rtree/0.8.3 py-scikit-learn/0.23.2 python-dev/0.4 py-virtualenv/16.7.6 asciitoh5/1.0 brainbuilder/0.16.1 brion/3.1.0 brion/3.2.0 brion/3.3.0 brion/3.3.1 circuit-build/3.1.3 connectome-tools/0.5.0 model-neocortex/1.1 nest/2.18.0 nest/2.20.0 neurodamus-core/3.3.0 neurodamus-core/3.3.0-commonmods neurodamus-hippocampus/1.5-3.3.0 neurodamus-hippocampus-knl/1.5-3.3.0 neurodamus-mousify/1.4-3.3.0 neurodamus-neocortex/1.4-3.3.0 neurodamus-neocortex-knl/1.4-3.3.0 neurodamus-thalamus/1.4-3.3.0 neurodamus-thalamus-knl/1.4-3.3.0 parquet-converters/0.5.7 placement-algorithm/2.1.2 psp-validation/0.4.0 py-atlas-building-tools/0.1.1 py-basalt/0.2.9 py-bbp-analysis-framework/1.6.42 py-bbp-workflow/2.1.14 py-bglibpy/4.4.36 py-bluepy/2.3.0 py-bluepyefe/0.3.13 py-bluepyemodel/0.0.3 py-bluepymm/0.7.65 py-bluepyopt/1.9.37 py-bluepyparallel/0.0.5 py-bluepysnap/0.12.0 py-currentscape/0.0.10 py-efel/3.0.80 py-entity-management/1.2.10 py-minis-validation/0.0.5 py-morphio/3.1.1 py-morphology-repair-workflow/2.0.3 py-morph-tool/2.5.0 py-neurodamus/2.6.0 py-notebook/6.1.4 py-pytouchreader/1.4.6 py-region-grower/0.2.2 py-simwriter/1.1.0 py-sonata-network-reduction/0.1.0 py-visual-cortex-analysis/develop regiodesics/0.1.2 spatial-index/0.2.1 spykfunc/0.16.0 spykfunc/0.16.0-spark3.0 steps/3.6.0 unit-test-translator/0.0.1
[2021-06-04T13:03:09.371Z] --------------------------
[2021-06-04T13:03:09.371Z] Module spatial-index/0.2.1
[2021-06-04T13:03:09.371Z] --------------------------
[2021-06-04T13:03:09.371Z] spatial-index 0.2.1 requires numpy, which is not installed.
[2021-06-04T13:03:09.371Z] make: *** [spatial-index/0.2.1] Error 1
[2021-06-04T13:03:21.733Z] ----------------------------------------
[2021-06-04T13:03:21.733Z] Module py-visual-cortex-analysis/develop
[2021-06-04T13:03:21.733Z] ----------------------------------------
[2021-06-04T13:03:21.733Z] bmtk 0.0.8 requires scikit-image, which is not installed.
[2021-06-04T13:03:21.733Z] make: *** [py-visual-cortex-analysis/develop] Error 1
[2021-06-04T13:03:26.009Z] ------------------------------------
[2021-06-04T13:03:26.009Z] Module py-atlas-building-tools/0.1.1
[2021-06-04T13:03:26.009Z] ------------------------------------
[2021-06-04T13:03:26.009Z] atlas-building-tools 0.1.1 requires scikit-image, which is not installed.
[2021-06-04T13:03:26.009Z] make: *** [py-atlas-building-tools/0.1.1] Error 1
[2021-06-04T13:03:36.117Z] ----------------------------
[2021-06-04T13:03:36.117Z] Module py-bluepyemodel/0.0.3
[2021-06-04T13:03:36.117Z] ----------------------------
[2021-06-04T13:03:36.117Z] bluepyemodel 0.0.3 requires gitpython, which is not installed.
[2021-06-04T13:03:36.117Z] bluepyemodel 0.0.3 requires psycopg2, which is not installed.
[2021-06-04T13:03:36.117Z] make: *** [py-bluepyemodel/0.0.3] Error 1
[2021-06-04T13:03:36.117Z] + echo 'Removing temporary files'
[2021-06-04T13:03:36.117Z] Removing temporary files
[2021-06-04T13:03:36.117Z] + rm -rf /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/1187/tmp/tmp.KIuNMtUtI6/venv /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/1187/tmp/tmp.YFvZB3EhyM
script returned exit code 2
```

I don't think this is coming from this PR.

Any help @GianlucaFicarelli @matz-e ? cc: @ferdonline

**Comment by ferdonline** [comment]
I don't think this is related either. This code path is not even executed from the CI, only when `+ngv` is requested from the ngv environment builder.
After the checks by Gianluca we are probably gonna see a few such errors to fix. From our side we can start with spatial index which seems to have missing dependency specification.

**Comment by GianlucaFicarelli** [comment]
Maybe it's better to clean the build workspace?
The `spatial-index` dependency was fixed in #1189

**Comment by matz-e** [comment]
Rebasing on latest develop should help, too?

**Comment by pramodk** [comment]
Thanks! Lets see how it goes!

**Review by alexsavulescu — APPROVED** [review]

**Pull request merged**